gen/api_defs.jl

```diff
@@ -205,6 +205,7 @@
 @bind h5s_get_simple_extent_dims(space_id::hid_t, dims::Ptr{hsize_t}, maxdims::Ptr{hsize_t})::Cint "Error getting the dimensions for a dataspace"
 @bind h5s_get_simple_extent_ndims(space_id::hid_t)::Cint "Error getting the number of dimensions for a dataspace"
 @bind h5s_get_simple_extent_type(space_id::hid_t)::Cint "Error getting the dataspace type"
+@bind h5s_get_select_npoints(space_id::hid_t)::hsize_t "Error getting the number of selected points"
```

Comment:
Adding a new API binding should also add another line into `docs/src/api_bindings.md` that should be committed as well.